public/javascript/index.js

```diff
@@ -1,23 +1,99 @@
-const charactersAPI = new APIHandler('http://localhost:8000');
+const charactersAPI = new APIHandler('https://minions-api.herokuapp.com/');
 
 window.addEventListener('load', () => {
   document.getElementById('fetch-all').addEventListener('click', function (event) {
 
+    charactersAPI
+      .getFullList()
+      .then(res => {
+        const { data } = res
+        let characters = ' '
+        data.reverse().forEach(elm => characters += `<div class="characters-container">
+                  <div class="character-info">
+                  <div> <strong> Name:</strong> <br> ${elm.name} (${elm.id})</div>
+                   <div> <strong> Occupation:</strong> <br> ${elm.occupation}</div>
+                   <div> <strong> Weapon:</strong> <br> ${elm.weapon}</div>
+                   <div> <strong> Is a cartoon? </strong><br> ${elm.cartoon}</div>
+                  </div>
+                  </div>`)
+        document.querySelector('.show-list').innerHTML = characters
+      })
+      .catch(err => console.log('Error:', err))
   });
 
   document.getElementById('fetch-one').addEventListener('click', function (event) {
+    const fetchOne = document.querySelector('#character-id').value
+
+    charactersAPI
+      .getOneRegister(fetchOne)
+      .then(res => {
+        const { data } = res
+        let singleCharacter = ''
+        singleCharacter += `<div class="characters-container">
+                  <ul class="character-info">
+                  <li> <strong> Name:</strong> <br> ${data.name}</li>
+                   <li> <strong> Occupation:</strong> <br> ${data.occupation}</li>
+                   <li> <strong> Weapon:</strong> <br> ${data.weapon}</li>
+                   <li> <strong> Is a cartoon? </strong><br> ${data.cartoon}</li>
+                  </ul>
+                  </div>`
+        document.querySelector('.show-list').innerHTML = singleCharacter
+      })
+      .catch(err => err, alert('Personaggio non esistente'))
```

Comment:
```suggestion
      .catch(err => { 
         console.log(err)
         alert('Personaggio non esistente')
      })
```
Si no empleas el error para nada puedes quitarlo porque no aporta al programa. Ese alert separado con una coma confunde a javascript y dispara el alert SIEMPRE que pulsas el botón.

Si quieres tener más de una instrucción en una arrow function usa el bloque.

Reply:
Vale! muchas gracias!